tcp_connection/test_tcp_connection.py

```diff
@@ -0,0 +1,167 @@
+"""
+Tests for TCP connection
+"""
+
+from typing import Optional
+
+from scapy.layers.inet import TCP
+from scapy.packet import Raw
+
+from helpers.tcp_client import (
+    ScapyLocalhostRequestFix,
+    ScapyTCPHandshakeResetFix,
+    SimpleTCPClient,
+)
+from test_suite import tester
+
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2024 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+
+class TesTCPConnection(tester.TempestaTest):
+    tempesta = {
+        "config": """
+            cache 0;
+            listen 80;
+            server ${server_ip}:8000;
+        """
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.tcp_client: Optional[SimpleTCPClient] = None
+
+    def setUp(self):
+        super().setUp()
+
+        self.start_tempesta()
+        self.start_all_servers()
+        self.deproxy_manager.start()
+        self.assertTrue(self.wait_all_connections())
+
+        tempest_host = self.get_tempesta()
+        self.tcp_client = SimpleTCPClient(
+            destination_host=tempest_host.host,
+            destination_port=80,
+            timeout=1,
+        )
+        self.tcp_client.fixes_install(ScapyTCPHandshakeResetFix, ScapyLocalhostRequestFix)
+
+    def tearDown(self):
+        super().tearDown()
+        self.tcp_client.fixes_rollback()
+
+    def test_tcp_connection_open_and_close_successfully(self):
+        """
+        Open and close tcp connection
+        """
+        init_request, init_response = self.tcp_client.request(flags="S")
+        self.assertIsNotNone(init_response)
+        self.assertEqual(init_response[TCP].flags, "SA")
```

Comment:
I would like to have a wrapper for IP from `scapy`. It looks uncomfortable when you don't know what you can get from the package.  And I would not like to receive errors from `scapy` that the packet has not a TCP layer.